File: iniciante/python/1046-tempo-de-jogo.py
def tempo_jogo(tempo_inicial, tempo_final):
    if tempo_inicial < tempo_final:
        tempo_jogado = tempo_final - tempo_inicial
    elif tempo_inicial > tempo_final:
        tempo_jogado = 24 - tempo_inicial + tempo_final
    else:
        tempo_jogado = 24
    return print(f'O JOGO DUROU {tempo_jogado} HORA(S)')


def entrada():
    tempos = list(map(int, input().split(' ')))
    horario_inicial = tempos[0]
    horario_final = tempos[1]
    return horario_inicial, horario_final


inicial, final = entrada()
tempo_jogo(inicial, final)

File: iniciante/c/1046-tempo-de-jogo.c
#include <stdio.h>

int main(){
    int hora_inicial, hora_final, tempo_de_jogo;
    
    scanf("%d %d", &hora_inicial, &hora_final);

    if (hora_final > hora_inicial){
        tempo_de_jogo = hora_final - hora_inicial;
    }
    else if (hora_final < hora_inicial){
        tempo_de_jogo = 24 + hora_final - hora_inicial;
    }
    else{
        tempo_de_jogo = 24;
    }
    printf("O JOGO DUROU %d HORA(S)\n", tempo_de_jogo);
    
    return 0;
}
